src/index.js

```diff
@@ -1,22 +1,40 @@
-import React, { useState } from "react";
-import AppNavigator from "./navigation/navigators/AppNavigator";
+import React, { useState, useEffect } from "react";
+import AsyncStorage from "@react-native-community/async-storage";
+import {
+  AppExtendedContainer,
+  AppPrimaryContainer
+} from "./navigation/navigators/AppNavigator";
 import TranslationProvider, {
   Translation
 } from "./components/TranslationProvider";
 import { changeLocale } from "./i18n";
 
-const AppContainer = () => {
+const AppLocalised = () => {
   const [locale, changeLocaleState] = useState("ru");
+  const [isFirstLaunch, toggleFirstLaunch] = useState(true);
+  let UltimateNavigator = AppExtendedContainer;
+
   const onLocaleChange = localeKey => {
     changeLocale(localeKey);
     changeLocaleState(localeKey);
```

Comment:
Basically here you don't need too changeLocale calls.
When calling changeLocale it will emit event, which will be caught by TranslationProvide, it will update locale prop inside itself, so all inner components will get it.
<TranslationProvider locale={locale}> - locale here it's just default initial locale

Reply:
Even if TranslationProvider updates the locale on app container, _locale_ prop is used to set state on language radio buttons in settings (here's the [screen](https://epam.invisionapp.com/d/main#/console/17142229/362606213/preview))